src/Compilers/Core/MSBuildTask/ArgsTaskItem.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    internal sealed class ArgsTaskItem : ITaskItem
```

Comment:
Can we add a comment summarizing that we're using this to avoid unnecessary path escaping?